### DISCUSSION TITLE
Exporting Rows to Read/Write  Custom UI Displays

### DISCUSSION THREAD

**Discussion by Mcc83** [Budibase/budibase#12673] [Help] [answered]
I am struggling to find the right path forward here and would appreciate some help.

My company is looking to export rows to an expanded view. Our tables are 17 columns wide and what we really need is a way to search a table (which Ive accomplished), send/export the data (I've read up on row selection and export), and then a way to display the data in a more user friendly UI. I would build out a form if I could figure out a way to send the data to it.

I thought about doing this through cards but I'm unsure if you can customize the buildout of the card into an expanded view if you pull a row from a table. However I recognize that cards are probably the wrong way to go about it.

The guides and videos on data views really only cover tables, cards, and graphs. Non of which are helping me AFAIK.

This is something I'll be expanding into a few other features such as making invoices/receipts. What seems to be everyones favorite method via automation or buttons to send & display rows into better views? Is there a way to allow read/write in forms so that our users can search and edit the database through a more friendly process outside of a mega table? Sorry if this is a bad question, I am very new to this and did my best effort of the last few days searching.

Thanks

**Comment by melohagan** [comment, marked as answer]
Hey @Mcc83 

If you use the [Table block](https://docs.budibase.com/docs/table-block) you can add up to five search columns. If 17 columns is too much in one screen, you could [Configure columns](https://docs.budibase.com/docs/table#configure-columns) to show the most important ones, and then on row click show a detailed side panel with all the info.

**Comment by Mcc83** [comment]
Hey @melohagan 

Thanks for the helpful answer. Is there anyway to customize the table block... into not a table? (probably a bad question). Could I display the table in an expanded view or perhaps send a row of data to a UI. 

What I really need is a search function for my beautiful forms which I could ideally do via the form block but AFAIK that isn't possible (I would even be satisfied if I could do it with cards  because I could painfully customize the cards and just search those?) 

The table block does indeed allow me to filter columns/data but what I need is 1 row of data to be pulled into an expanded UI. Ideally it's displayed the exact same way as my form look which I am happy to build myself, just want to know if it's possible. I'm sure I am not being great with my explanation. But a table view is just not what I am looking for. Is there anyway to maybe search a tableblock and export to a UI page? That's the only feasible way I see this working from my limited understanding.

My users don't want a row of data, even if its somewhat cut down, they need a nice big ol' readout where the data is sent.

**Reply by melohagan** [reply]
You can build custom searching functionality with a form field which is bound to a filtered data provider/repeater. There's an example here: https://docs.budibase.com/docs/adding-a-search-field#2-using-a-form-and-text-input-flexible-way
